internal/controllers/machineset/machineset_controller.go

```diff
@@ -987,6 +1047,26 @@ func (r *Reconciler) reconcileUnhealthyMachines(ctx context.Context, cluster *cl
 		return ctrl.Result{}, nil
 	}
 
+	// Sort the machines from newest to oldest.
+	// We are trying to remediate machines failing to come up first because
+	// there is a chance that they are not hosting any workloads (minimize disruption).
+	sort.Slice(machinesToRemediate, func(i, j int) bool {
+		return machinesToRemediate[i].CreationTimestamp.After(machinesToRemediate[j].CreationTimestamp.Time)
+	})
+
+	// At this point we have a list of machines that need to be remediated, and the number of tokens available to do so.
+	// Check if we can remediate any machines.
+	if maxInFlight <= 0 {
```

Comment:
I would move this up directly below l.1048 and just return in that case (so we can get rid of the else branch, and we don't need the sort above)

Reply:
done